### PR TITLE
updated server with mongodb and reworked files

### PR DESCRIPTION
**Pull request by kdsolo**
Resolves #146 (Store data in MongoDB)

### CODE REVIEW

File: haiku-data.json [hunk 1]
```diff
@@ -0,0 +1,182 @@
+{
```
Comment (sfoster) [inline haiku-data.json:1]: Was there a script that created this file?

File: package.json [hunk 2]
```diff
@@ -19,8 +21,8 @@
     "lint"
   ],
   "scripts": {
-    "start": "nodemon src/index.js",
-    "lint": "semistandard src/index.js",
+    "start": "nodemon src/server/index.js",
```
Comment (sfoster) [inline package.json:24]: Why did we add this 'server' directory. This is the repo for this server, are we expecting other kinds of src-s. It seems redundant

File: README.md [hunk 3]
```diff
@@ -13,6 +13,22 @@ $ npm install
 ```
 
 
+## Starting Database
+
+Be sure to install MongoDB on your computer and start up the database using:
```
Comment (sfoster) [inline README.md]: Can you add a bit more detail on where/how to install mongodb? Maybe just links to https://docs.mongodb.com/manual/administration/install-on-linux/ and https://docs.mongodb.com/manual/tutorial/install-mongodb-on-os-x/

**Comment by sfoster** [comment]
This looks great. If you can address the comments - moving this back into the src directory and linux + osx mongodb install links - I think we'll be good. I'll take it for a spin later and report back.

File: haiku-data.json (hunk 1 above)
Comment (kdsolo) [inline haiku-data.json:1]: No I wrote it myself but I can look into building a script or finding one online if we need it.

File: README.md (hunk 3 above)
Comment (kdsolo) [inline README.md]: Yes I'll add those links and more description

File: package.json (hunk 2 above)
Comment (kdsolo) [inline package.json:24]: I split up the route handler code (index.js) into its own files, so now there are multiple files on the server side so it seems more organized to keep the server files separate from the public files

Comment (sfoster) [reply package.json:24]: I also talked with Reza about this and agreed that separating the server from the client code is a useful step

File: README.md [hunk 4]
```diff
@@ -13,6 +13,31 @@ $ npm install
...
+$ mongod
+```
+
+
+With `haiku-data.json` in your current working directory run this command to import the data:
+
+```
+$ mongoimport --db haiku --collection users --drop --file haiku-data.json
+```
+
```
Comment (sfoster) [inline README.md:39]: I'd like to see more in here about working with the db - maybe mongo shell examples for the basic CRUD operations we were previously using a text/code editor for. But lets do this is a follow-up.

File: src/index.js [hunk 5]
```diff
@@ -1,126 +1,42 @@
...
-    }
-
-    res.setHeader('Content-Type', 'application/json');
-    res.send(buf.toString());
-  });
+mongoose.connect('mongodb://localhost/haiku');
+var db = mongoose.connection;
+db.on('error', console.error.bind(console, 'connection error:'));
+db.once('open', function() {
+  console.log("Connected to UsersDB with Mongoose");
```
Comment (sfoster) [inline src/index.js]: "Connected to 'haiku' database with Mongoose"?

**Comment by sfoster** [comment]
Thanks for splitting out the refactor and the directory restructuring, that's now much clearer. 
I did the install and data import and took this for a quick spin. If you rebase on master, you'll see a couple things that need fixing: 
- Procfile - this is the file used by Heroku to know what to run and how. It needs the src/index.js path fixing. With that, I think this is ready to land. 
- bin/createUser.js - was the quickie script I made to create new users for Experiment #4. We'll almost certainly want to do that again, so while we don't need to address it in this PR, lets file a new bug to migrate this functionality and try and get it in soon

**Comment by sfoster** [comment]
Looks good @kdsolo, can you file the 2 follow up issues - for the mongo shell CRUD examples, and for the createUser.js replacement.

**Pull request merged**